nessie/src/test/java/org/apache/iceberg/nessie/TestNessieCatalog.java

```diff
@@ -70,12 +72,14 @@ public class TestNessieCatalog extends CatalogTests<NessieCatalog> {
   private String uri;
 
   @BeforeEach
-  public void beforeEach(@NessieUri URI nessieUri) throws IOException {
-    this.uri = nessieUri.toString();
-    this.api = HttpClientBuilder.builder().withUri(this.uri).build(NessieApiV1.class);
-
-    initialHashOfDefaultBranch = api.getDefaultBranch().getHash();
-
+  public void setUp(NessieClientFactory clientFactory, @NessieClientUri URI nessieUri) {
+    api = clientFactory.make();
+    try {
+      initialHashOfDefaultBranch = api.getDefaultBranch().getHash();
+    } catch (NessieNotFoundException e) {
```

Comment:
Why not declare the exception as thrown by this method? JUnit 5 can deal with that and the code will be simpler and easier to read :thinking:

Reply:
done